treepoem/__main__.py

```diff
@@ -0,0 +1,35 @@
+# -*- encoding:utf-8 -*-
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import argparse
+import sys
+from textwrap import fill
+
+from . import generate_barcode
+from .data import barcode_types
+
+supported_barcode_types = (
+    'Supported barcode types are:\n'
+    + fill(', '.join(sorted(barcode_types)), initial_indent='    ', subsequent_indent='    ')
+)
+
+
+def main():
+    p = argparse.ArgumentParser(epilog=supported_barcode_types)
+    p.add_argument('-t', '--type', default='qrcode', help='Barcode type (default (%(default)s))')
+    p.add_argument('-f', '--fmt', help='Output format (default is based on file extension)')
+    p.add_argument('-o', '--output', help='Output file (default is stdout)')
+    p.add_argument('data', help='Barcode data')
+    p.add_argument('options', nargs='*', type=lambda x: x.split('=', 1)[:2], help='List of BWIPP options (e.g. width=1.5)')
+    args = p.parse_args()
+
+    if args.type not in barcode_types:
+        p.error('Barcode type %r is not supported. %s' % (args.type, supported_barcode_types))
+
+    if args.output is None:
+        args.output = sys.stdout.buffer
```

Comment:
can't this be done with the `default` argument of `add_argument` ?

Also this doesn't work on Python 2

```
In [2]: sys.stdout.buffer
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-8a9e92bf94df> in <module>()
----> 1 sys.stdout.buffer

AttributeError: 'file' object has no attribute 'buffer'
```

Tests would have caught that

Reply:
Good point. The problem is that `sys.stdout.write(b'byte_string')` won't work on 3.x. I'm not sure what the best way to handle it is… maybe just a version check?

The reason I didn't use `default=sys.stdout` is that this results in an annoying inconsistency, wherein the _default_ value for the argument is a file object, but the user-input values will be strings containing filenames. `PIL` is very good at guessing the output type from a filename, but not from a file object.

I'll tweak this a bit and see what you think…

Reply:
you can do `if hasattr(sys.stdout, 'buffer')`